### PR TITLE
Add alias links to web UI

### PR DESCRIPTION
**Pull request by pascalwengerter**
## Description
~Cleaning up my local git stash and came across this draft. Needs concepting how to properly turn this into an oCIS-only feature.~ Update: alias-link capability has landed in oCIS

## Related Issue
- Works towards #6844

### CODE REVIEW

**Comment by pascalwengerter** [comment]
This does now work on the link-creator side, but I run into problems when trying to open the (permissionless) link - I'd expect it to render the login form, but it shows an error page (with a 401 when listing drives)

**Comment by kulmann** [comment]
@pascalwengerter I took the liberty of rebasing this PR, as the public link resolving has changed in the web runtime today (see https://github.com/owncloud/web/pull/7072 for reference). However, whenever I want to create an internal link I get `invalid permissions` from the backend. Can you confirm that?

**Comment by pascalwengerter** [comment]
> > @pascalwengerter I took the liberty of rebasing this PR, as the public link resolving has changed in the web runtime today (see #7072 for reference). However, whenever I want to create an internal link I get `invalid permissions` from the backend. Can you confirm that?
> 
> Did some investigation now and can confirm it doesn't work - I thought we had implemented the necessary changes in the SDK already, but getting an `invalid permissions` webdav response when trying to sending e.g. a "0" as zero-permission (sending _nothing_ doesn't work because that is the standard behavior when updating links without touching permissions). I've opened [an issue in the SDK](https://github.com/owncloud/owncloud-sdk/issues/1114)

Turns out sending a `0` permission is enough, it does work with https://github.com/cs3org/reva/pull/3079 (stopped CI, this PR needs a rebase & review once Julian's PR is merged in reva and then in oCIS and the oCIS commit ID is bumped)!

**Comment by pascalwengerter** [comment]
> > > @pascalwengerter I took the liberty of rebasing this PR, as the public link resolving has changed in the web runtime today (see #7072 for reference). However, whenever I want to create an internal link I get `invalid permissions` from the backend. Can you confirm that?
> > 
> > 
> > Did some investigation now and can confirm it doesn't work - I thought we had implemented the necessary changes in the SDK already, but getting an `invalid permissions` webdav response when trying to sending e.g. a "0" as zero-permission (sending _nothing_ doesn't work because that is the standard behavior when updating links without touching permissions). I've opened [an issue in the SDK](https://github.com/owncloud/owncloud-sdk/issues/1114)
> 
> Turns out sending a `0` permission is enough, it does work with [cs3org/reva#3079](https://github.com/cs3org/reva/pull/3079) (stopped CI, this PR needs a rebase & review once Julian's PR is merged in reva and then in oCIS and the oCIS commit ID is bumped)!

Ideally we also get an E2E test for alias links in oCIS to make sure they don't break unnoticed in the near future

**Comment by kulmann** [comment]
This PR handles **creating** internal / permissionless links, that works with updated ocis/reva as you mentioned. Respective commit id has been bumped in web already, so a rebase was sufficient. But: resolving an internal link is not handled in the public link resolve page. I'd like to merge this PR but flip the default of the capability to internal links being disabled, until we implement resolving internal links in a separate PR. Do you agree @pascalwengerter ?

**Comment by pascalwengerter** [comment]
> This PR handles **creating** internal / permissionless links, that works with updated ocis/reva as you mentioned. Respective commit id has been bumped in web already, so a rebase was sufficient. But: resolving an internal link is not handled in the public link resolve page. I'd like to merge this PR but flip the default of the capability to internal links being disabled, until we implement resolving internal links in a separate PR. Do you agree @pascalwengerter ? 

Please do push to this PR as you deem necessary, I don't have any preference on how to proceed here ;)

**Comment by JammingBen** [comment]
I rebased and updated the PR to go along with current master. No idea why the condition coverage does not cover `availableRoleOptions()` though. Locally, the tests call this method and reach the updated return statement 🤷 

Besides that, this should be ready for review. LGTM to me, although I'm not too invested in the topic.

File: changelog/unreleased/enhancement-permissionless-links [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Enhancement: Permissionsless (internal) link shares
```
Comment (kulmann) [inline changelog/unreleased/enhancement-permissionless-links:1]:
```suggestion
Enhancement: Permissionless (internal) link shares
```

File: packages/web-app-files/src/components/SideBar/Shares/FileLinks.vue [hunk 2]
```diff
@@ -219,15 +219,11 @@ export default defineComponent({
         )
       }
 
-      const roles = LinkShareRoles.list(
+      return LinkShareRoles.list(
         this.highlightedFile.isFolder,
-        this.capabilities.files_sharing?.public?.can_edit
+        this.capabilities.files_sharing?.public?.can_edit,
```
Comment (kulmann) [inline packages/web-app-files/src/components/SideBar/Shares/FileLinks.vue:224]: Let's introduce capability composables for these two capabilities (look for `useCapability...` for examples).

**Review by kulmann — CHANGES_REQUESTED** [review]
Some nitpicks (see comments). Also: please update the ocis commit id in `.drone.env` to one after the alias capability value has changed in ocis.

**Comment by JammingBen** [comment]
Arghs I forgot the `.drone.env`, coming up next

**Review by kulmann — APPROVED** [review]

**Pull request merged**